python-libraries/nanover-ase/src/nanover/ase/imd.py

```diff
@@ -311,6 +307,38 @@ def reset(self):
         self.atoms.set_cell(self._initial_box)
         self._call_on_reset()
 
+    def replace_dynamics(self, dynamics: MolecularDynamics, reset=True):
+        if reset:
```

Comment:
Do we document the reset event anywhere? The `on_reset_listener`? It should document that when a new simulation is loaded, the event is called from the old simulation, based on where the reset call is located.

Reply:
Not sure. I don't know what it was intended for, but it seems it happens either when the user chooses to reset or when the simulation explodes and resets itself. The two places it's used are in the CLI to print loudly that a reset happened, and in the trajectory logger where resets start a new recording file.

BTW do you see any reason why there should be a difference between a reset and reloading the current simulation?

Reply:
If we just loudly yell reset we should be good whatever simulation is current when the event occurs.

There is a small semantic difference between reset and reload. When resetting, you do not expect the topology to change while you expect it when loading a new simulation. Sending a new topology will affect `nanover.mdanalysis` when reading recordings since it stops whenever it receives a new topology.